docs/azure/sdk/authentication/authentication-best-practices.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Authentication best practices with the Azure Identity library for .NET
+description: This article describes authentication best practices to follow when using the Azure Identity library for .NET.
+ms.topic: conceptual
+ms.date: 01/15/2025
+---
+
+# Authentication best practices with the Azure Identity library for .NET
+
+This article offers guidelines to help you maximize the performance and reliability of your .NET apps when authenticating to Azure services. To make the most of the Azure Identity library for .NET, it's important to understand potential issues and mitigation techniques.
+
+## Use deterministic credentials in production environments
+
+[!INCLUDE [default-azure-credential-usage](../includes/default-azure-credential-usage.md)]
+
+## Reuse credential instances
+
+Reuse credential instances when possible to improve app resilience and reduce the number of access token requests issued to Microsoft Entra ID. When a credential is reused, an attempt is made to fetch a token from the app token cache managed by the underlying MSAL dependency. For more information, see [Token caching in the Azure Identity client library](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/samples/TokenCache.md).
+
+> [!NOTE]
+> A high-volume app that doesn't reuse credentials may encounter HTTP 429 throttling responses from Microsoft Entra ID, which can lead to app outages.
+
+In an ASP.NET Core app, implement credential reuse through the `UseCredential` method of `Microsoft.Extensions.Azure`:
+
+:::code language="csharp" source="../snippets/authentication/best-practices/Program.cs" id="snippet_credential_reuse_Dac" highlight="6" :::
+
+For information on this approach, see [Authenticate using Microsoft Entra ID](/dotnet/azure/sdk/aspnetcore-guidance?tabs=api#authenticate-using-microsoft-entra-id).
+
+Other types of .NET apps can reuse credential instances as follows:
+
+:::code language="csharp" source="../snippets/authentication/best-practices/Program.cs" id="snippet_credential_reuse_noDac" highlight="8, 12" :::
```

Comment:
We should tidy this up by introducing 2 tabs: "ASP.NET Core" and "Other". That change reduces the vertical scrolling.